ActionScript/ActionScript.sublime-syntax

```diff
@@ -12,87 +12,892 @@ first_line_match: |-
-      scope: support.constant.actionscript.2
-    - match: \b(dynamic|extends|import|implements|interface|public|private|new|static|super|var|for|in|break|continue|while|do|return|if|else|case|switch)\b
-      scope: keyword.control.actionscript.2
-    - match: \b(Boolean|Number|String|Void)\b
-      scope: storage.type.actionscript.2
-    - match: \b(null|undefined|true|false)\b
-      scope: constant.language.actionscript.2
-    - match: '\b((0(x|X)[0-9a-fA-F]*)|(([0-9]+\.?[0-9]*)|(\.[0-9]+))((e|E)(\+|-)?[0-9]+)?)(L|l|UL|ul|u|U|F|f)?\b'
-      scope: constant.numeric.actionscript.2
-    - match: '"'
-      captures:
-        0: punctuation.definition.string.begin.actionscript.2
-      push:
-        - meta_scope: string.quoted.double.actionscript.2
-        - match: '"'
-          captures:
-            0: punctuation.definition.string.end.actionscript.2
-          pop: true
-        - match: \\.
-          scope: constant.character.escape.actionscript.2
-    - match: "'"
-      captures:
-        0: punctuation.definition.string.begin.actionscript.2
-      push:
-        - meta_scope: string.quoted.single.actionscript.2
-        - match: "'"
-          captures:
-            0: punctuation.definition.string.end.actionscript.2
-          pop: true
-        - match: \\.
-          scope: constant.character.escape.actionscript.2
-    - match: \b(BACKSPACE|CAPSLOCK|CONTROL|DELETEKEY|DOWN|END|ENTER|HOME|INSERT|LEFT|LN10|LN2|LOG10E|LOG2E|MAX_VALUE|MIN_VALUE|NEGATIVE_INFINITY|NaN|PGDN|PGUP|PI|POSITIVE_INFINITY|RIGHT|SPACE|SQRT1_2|SQRT2|UP)\b
-      scope: support.constant.actionscript.2
-    - match: /\*
-      captures:
-        0: punctuation.definition.comment.actionscript.2
-      push:
-        - meta_scope: comment.block.actionscript.2
-        - match: \*/
-          captures:
-            0: punctuation.definition.comment.actionscript.2
-          pop: true
-        - match: ^\s*(\*)(?!/)
-          captures:
-            1: punctuation.definition.comment.actionscript.2
-    - match: (//).*$\n?
-      scope: comment.line.double-slash.actionscript.2
-      captures:
-        1: punctuation.definition.comment.actionscript.2
-    - match: \b(instanceof)\b
-      scope: keyword.operator.actionscript.2
-    - match: "[-!%&*+=/?:]"
-      scope: keyword.operator.symbolic.actionscript.2
+    - match: \b(?:{{classes}})\b
+      scope: support.class.actionscript
+
+    - match: \b(?:{{functions}})\b
+      scope: support.function.actionscript
+
+    - match: \b(?:{{constants}})\b
+      scope: support.constant.actionscript
+
+    - match: \b(?:{{keywords}})\b
+      scope: keyword.control.actionscript
+
+    - match: \b(?:{{types}})\b
+      scope: storage.type.actionscript
+
+    - include: constants
+
+    - include: strings
+
+    - match: \b(?:instanceof)\b
+      scope: keyword.operator.actionscript
+
+    - match: '[-!%&*+=/?:]'
+      scope: keyword.operator.symbolic.actionscript
```

Comment:
Same here. While `keyword.operator.symbolic` is neat with regards to distinguish those from word-like operators, it doesn't follow currently applied scopes of all other syntaxes. Would be a good chance to change that.